tests/src/app/testqgsmeasuretool.cpp

```diff
@@ -131,6 +131,24 @@ void TestQgsMeasureTool::testLengthCalculation()
   measured = measureString.remove( ',' ).split( ' ' ).at( 0 ).toDouble();
   expected = 88360.0918635;
   QVERIFY( qgsDoubleNear( measured, expected, 0.001 ) );
+
+  // check new CoordinateReferenceSystem, points must be reprojected to paint them successfully (issue #15182)
+  QgsCoordinateReferenceSystem srs2( 4326, QgsCoordinateReferenceSystem::EpsgCrsId );
+
+  QgsCoordinateTransform ct( srs, srs2 );
+
+  QgsPoint p0 = ct.transform( tool2->points()[0] );
+  QgsPoint p1 = ct.transform( tool2->points()[1] );
+
+  mCanvas->setDestinationCrs( srs2 );
+
+  QgsPoint n0 = tool2->points()[0];
+  QgsPoint n1 = tool2->points()[1];
+
+  QVERIFY( qgsDoubleNear( p0.x(), n0.x(), 0.001 ) );
```

Comment:
There's now `QGSCOMPARENEAR` which is nicer if it comes to debugging

https://github.com/qgis/QGIS/commit/e95a8a979a62fdd09fd678efafe28f95b44ff50c#diff-9aafa4a9dec16b6386ec6feacf94e2f9R608